kms/src/main/java/io/kroxylicious/kms/provider/kroxylicious/inmemory/InMemoryKms.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.kroxylicious.inmemory;
+
+import java.security.GeneralSecurityException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+
+import io.kroxylicious.kms.service.De;
+import io.kroxylicious.kms.service.DekPair;
+import io.kroxylicious.kms.service.Kms;
+import io.kroxylicious.kms.service.KmsException;
+import io.kroxylicious.kms.service.Ser;
+import io.kroxylicious.kms.service.UnknownAliasException;
+import io.kroxylicious.kms.service.UnknownKeyException;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public class InMemoryKms implements
+        Kms<UUID, InMemoryEdek> {
+
+    private static final String WRAP_ALGO = "AES_256/GCM/NoPadding";
+    public static final String KEY_ALGO = "AES";
+    private final Map<UUID, SecretKey> keys;
+    private final KeyGenerator aes;
+    private final int numIvBytes;
+    private final int numAuthBits;
+    private final SecureRandom secureRandom;
+    private final Map<String, UUID> aliases;
+
+    InMemoryKms(int numIvBytes, int numAuthBits,
+                Map<UUID, SecretKey> keys,
+                Map<String, UUID> aliases) {
+        this.keys = keys;
+        this.aliases = aliases;
+        this.secureRandom = new SecureRandom();
+        this.numIvBytes = numIvBytes;
+        this.numAuthBits = numAuthBits;
+        try {
+            this.aes = KeyGenerator.getInstance(KEY_ALGO);
+        }
+        catch (NoSuchAlgorithmException e) {
+            // This should be impossible, because JCA guarantees that AES is available
+            throw new KmsException(e);
+        }
+    }
+
+    /**
+     * Generates a KEK
+     * @return The id of the KEK
+     */
+    public UUID generateKey() {
+        var key = aes.generateKey();
+        var ref = UUID.randomUUID();
+        keys.put(ref, key);
+        return ref;
+    }
+
+    public void createAlias(UUID keyId, String alias) {
+        lookupKey(keyId); // check the key exists in this KMS
+        aliases.put(alias, keyId);
+    }
+
+    public void deleteAlias(String alias) {
+        var was = aliases.remove(alias);
+        if (was == null) {
+            throw new UnknownAliasException(alias);
+        }
+    }
+
+    @NonNull
+    @Override
+    public CompletableFuture<InMemoryEdek> generateDek(@NonNull UUID kekRef) {
```

Comment:
The return type should be CompletionStage.

Reply:
This is for testing, where it's slightly more convenient to return a `CompletableFuture`, on which we can call `get` directly.